Src/Base/AMReX.cpp

```diff
@@ -538,16 +555,30 @@ amrex::Initialize (int& argc, char**& argv, bool build_parm_parse,
     iMultiFab::Initialize();
     VisMF::Initialize();
     AsyncOut::Initialize();
+
 #ifdef AMREX_USE_EB
     EB2::Initialize();
 #endif
+
     BL_PROFILE_INITPARAMS();
-#endif
+#endif // ifndef BL_AMRPROF
+
     machine::Initialize();
+
 #ifdef AMREX_USE_GPU
     Gpu::Fuser::Initialize();
 #endif
 
+#ifdef AMREX_USE_HYPRE
+    if (init_hypre) {
+        HYPRE_Init();
+#ifdef AMREX_USE_CUDA
+        hypre_HandleDefaultExecPolicy(hypre_handle()) = HYPRE_EXEC_DEVICE;
+        hypre_HandleSpgemmUseCusparse(hypre_handle()) = 0;
+#endif
+    }
+#endif
+
```

Comment:
```suggestion
#if defined(AMREX_USE_HYPRE) && defined(HYPRE_USING_CUDA)
    if (init_hypre) {
        HYPRE_Init();
        hypre_HandleDefaultExecPolicy(hypre_handle()) = HYPRE_EXEC_DEVICE;
        hypre_HandleSpgemmUseCusparse(hypre_handle()) = 0;
    }
#endif
```

@WeiqunZhang We should probably account for the fact that the user might attempt to link to a hypre that is not compiled for CUDA. I had used something like the one above to handle situations where hypre and AMReX were built differently. Do you have any thoughts on that?

Reply:
Thanks for the suggestion!  I have tested the current version using `HYPRE_USING_CUDA`.  It now works with hypre built with or without CUDA, when AMReX is built with CUDA.